File: Python/002. Add Two Numbers.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        carry = 0
        ans = ListNode(0)
        l = ans
        while l1 is not None and l2 is not None:
            num = l1.val + l2.val + carry
            carry = num // 10
            num = num % 10
            l.next = ListNode(num)
            l = l.next
            l1 = l1.next
            l2 = l2.next
        if l1 is None:
            p = l2
        else:
            p = l1
        while p is not None:
            num = p.val + carry
            carry = num // 10
            num = num % 10
            l.next = ListNode(num)
            l = l.next
            p = p.next
        if carry != 0:
            l.next = ListNode(carry)
        return ans.next

File: Go/002. Add Two Numbers.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	res := new(ListNode)
	p := res
	carry := 0
	for l1 != nil || l2 != nil || carry != 0 {
		if l1 != nil {
			carry += l1.Val
			l1 = l1.Next
		}
		if l2 != nil {
			carry += l2.Val
			l2 = l2.Next
		}
		p.Next = &ListNode{carry % 10, nil}
		p = p.Next
		carry = carry / 10
	}
	return res.Next
}